tutorials/pubsub-spring-dedup-messages/pubsub-spring/src/main/java/com/google/example/DataEntryController.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.example;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.messaging.Message;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.servlet.view.RedirectView;
+import reactor.core.publisher.EmitterProcessor;
+
+/**
+ * REST Controller for users to send messages.
+ */
+@RestController
+public class DataEntryController {
+
+  @Autowired
+  private EmitterProcessor<Message<String>> frontEndListener;
+
+  @PostMapping("/sendData")
+  public RedirectView mainPage(
+    @RequestParam("data") String data, @RequestParam("key") String key) {
+    System.out.println("Sending data: " + data + " of key " + key + "..");
```

Comment:
You might want to consider `String.format()` or `System.out.printf()`?

```
System.out.println(String.format("Sending data: {} of key {}", data, key));
System.out.printf("Sending data: %s of key %s", data, key);
```

They're all a bit inconsistent :/

Reply:
What's the benefit of switching to `String.format`?